### PR TITLE
Refactor BR exchanges

### PR DESCRIPTION
**Pull request by unitrium**
## Issue

Refactoring the BR exchanges to match the new event datastructure.

## Description

- upgrade the full BR parser to the new event datastructure for exchanges.
- merge all fetch_exchange functions into one.
- rename parser to match source name.

### Double check

- [ ] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [ ] I have run `pnpx prettier --write .` and `poetry run format` to format my changes.

### CODE REVIEW

File: parsers/ONS.py [hunk 1]
```diff
@@ -110,64 +102,36 @@ def fetch_production(
...
+def get_exchange_flow(sorted_zone_keys: ZoneKey, raw_data: dict) -> float:
+    """Returns the flow of the exchange between two regions."""
+    name, flow, level = EXCHANGES[sorted_zone_keys].values()
+    return raw_data[level][name] * flow
 
 
-def fetch_region_exchange(
+def fetch_exchange(
     zone_key1: str,
     zone_key2: str,
```
Comment (VIKTORVAV99) [inline parsers/ONS.py:113]: Should these not be typed to ZoneKey?

File: parsers/ONS.py [hunk 2]
```diff
@@ -192,13 +156,13 @@ def fetch_region_exchange(
...
-    print(fetch_region_exchange("BR-CS", "BR-NE"))
+    print(fetch_exchange("BR-CS", "BR-NE"))
 
     print("fetch_region_exchange(BR-CS->BR-N)")
-    print(fetch_region_exchange("BR-CS", "BR-N"))
+    print(fetch_exchange("BR-CS", "BR-N"))
 
     print("fetch_region_exchange(BR-N->BR-NE)")
-    print(fetch_region_exchange("BR-N", "BR-NE"))
+    print(fetch_exchange("BR-N", "BR-NE"))
```
Comment (VIKTORVAV99) [inline parsers/ONS.py:168]: It also means these need to be updated though to not introduce any type errors in the file. (the print statements containing the title should also be updated).

File: parsers/test/test_ONS.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 from arrow import get
```
Comment (VIKTORVAV99) [inline parsers/test/test_ONS.py:9]: Can we update the tests to also use datetime? the less arrow the better IMO.

File: parsers/ONS.py (hunk 1 above)
Comment (unitrium) [reply parsers/ONS.py:113]: Good catch, indeed they should.

**Review by VIKTORVAV99 — APPROVED** [review]
LGTM! 🎉

**Pull request merged**